### ISSUE TITLE
solo.predict() should take an argument to return Non-softmax transformed values, and return logits like previous iterations to allow for non-breaking change that was implemented in 1.1.3

### ISSUE THREAD

**Issue by barstuart** [scverse/scvi-tools#2869]
**Is your feature request related to a problem? Please describe.**
Previous fixes for solo (https://github.com/scverse/scvi-tools/pull/2689) and issue reported https://github.com/scverse/scvi-tools/issues/2688 , implemented a breaking change with no way to return logits. Some of us still use non-probability values in their pipelines. 

**Describe the solution you'd like**
I have implemented a quick fix for the above by creating a new argument for the solo.predict() function that allows users to return non-softmax normalized probabilities, and return the logits as in previous iterations, thus allowing a non-breaking fix in 1.1.3.

Here is the relevant code that i suggest be implemented

```

    @torch.inference_mode()
    def predict(self, soft: bool = True, include_simulated_doublets: bool = False, return_logits: bool = False) -> pd.DataFrame:
        """Return doublet predictions.

        Parameters
        ----------
        soft
            Return probabilities instead of class label.
        include_simulated_doublets
            Return probabilities for simulated doublets as well.
        return_logits
            Return logits instead of probabilities.

        Returns
        -------
        DataFrame with prediction, index corresponding to cell barcode.
        """
        warnings.warn(
            "Prior to scvi-tools 1.1.3, `SOLO.predict` with `soft=True` (the default option) "
            "returned logits instead of probabilities. This behavior has since been corrected to "
            "return probabiltiies.",
            UserWarning,
            stacklevel=settings.warnings_stacklevel,
        )

        adata = self._validate_anndata(None)
        scdl = self._make_data_loader(adata=adata)

        @auto_move_data
        def auto_forward(module, x):
            return module(x)

        y_pred = []
        for _, tensors in enumerate(scdl):
            x = tensors[REGISTRY_KEYS.X_KEY]
            logits = auto_forward(self.module, x)
            if return_logits:
                pred = logits
            else:
                pred = torch.nn.functional.softmax(logits, dim=-1)
            y_pred.append(pred.cpu())

        y_pred = torch.cat(y_pred).numpy()

        label = self.adata.obs["_solo_doub_sim"].values.ravel()
        mask = label == "singlet" if not include_simulated_doublets else slice(None)

        preds = y_pred[mask]

        cols = self.adata_manager.get_state_registry(REGISTRY_KEYS.LABELS_KEY).categorical_mapping
        preds_df = pd.DataFrame(preds, columns=cols, index=self.adata.obs_names[mask])

        if not soft and not return_logits:
            preds_df = preds_df.idxmax(axis=1)

        return preds_df



```

**Example usage**
`solo.predict()` will return a dataframe with probabilities (example below)

doublet | singlet
0.144867 | 0.855133

`solo.predict(soft=False)`  will return a dataframe with labels (doublet/singlet)

`solo.predict(return_logits=True)` will return a dataframe with logit values (as in versions <1.1.2) example below of same cell from above

doublet | singlet
-0.463635 | 1.311804

**Comment by martinkim0** [comment]
I think this change makes sense for reproducibility with previous versions, thanks for the suggestion. I'll add this change.

**Comment by martinkim0** [comment]
The change will be released with 1.1.4!
